frontend/src/pages/UserEditPage.jsx

```diff
@@ -1,14 +1,299 @@
+    if (validateForm()) {
+      setShowConfirmModal(true);
+      if (confirmModalRef.current) {
+        confirmModalRef.current.showModal();
+      }
+    }
+  };
+
+  const confirmUpdate = () => {
+    if (id && !isUpdating) {
+      updateUser(`/users/${id}`, {
+        headers: { 'content-type': 'application/json' },
+        body: JSON.stringify(formData),
+      });
+    }
+  };
+
+  const closeConfirmModal = () => {
+    setShowConfirmModal(false);
+    if (confirmModalRef.current) {
+      confirmModalRef.current.close();
+    }
+  };
+
+  if (isFetching) {
+    return <Spinny />;
+  }
+
+  if (fetchError && !fetchedUserData) {
+    return (
+      <div className='section-card flex flex-col items-center justify-center min-h-[60vh]'>
+        <h1 className='section-heading text-error'>Error</h1>
+        <p className='text-lg text-gray-600 mb-8'>
+          {fetchError?.message || 'Could not load user data for editing.'}
+        </p>
+        <button
+          onClick={() => navigate(`/users/${id}`)}
+          className='animated-button !text-base !px-5 !py-2'
+        >
+          Back to User Profile
+        </button>
+      </div>
+    );
+  }
+
+  if (!fetchedUserData) {
+    return (
+      <div className='section-card flex flex-col items-center justify-center min-h-[60vh]'>
+        <h1 className='section-heading'>User Not Found</h1>
+        <p className='text-lg text-gray-600 mb-8'>
+          The user you are trying to edit does not exist.
+        </p>
+        <button
+          onClick={() => navigate('/users')}
+          className='animated-button !text-base !px-5 !py-2'
+        >
+          Back to All Users
+        </button>
+      </div>
+    );
+  }
 
   return (
-    <div className='section-card flex flex-col items-center justify-center min-h-[60vh]'>
-      <h1 className='section-heading'>Edit User: {id}</h1>
-      <p className='text-lg text-gray-600 mb-8'>
-        This page will contain the form to edit the profile for user with ID:{' '}
-        {id}.
-      </p>
+    <div className='section-card flex flex-col items-center min-h-[60vh]'>
+      <h1 className='section-heading'>
+        Edit User: {formData.name || formData.username}
+      </h1>
+      <form
+        onSubmit={handleSubmit}
+        className='w-full max-w-xl p-4 bg-base-200 rounded-lg shadow-md'
+      >
+        <div className='mb-4'>
+          <label
+            htmlFor='username'
+            className='block text-lg font-medium text-gray-700 mb-2'
+          >
+            Username:
+          </label>
+          <input
+            type='text'
+            id='username'
+            name='username'
+            value={formData.username}
+            onChange={handleChange}
+            className='input input-bordered w-full p-2 rounded-md'
+            disabled={isUpdating}
+          />
+          {formErrors.username && (
+            <p className='text-error text-sm mt-1'>{formErrors.username}</p>
+          )}
+        </div>
```

Comment:
Should it be possible to change username? Api is currently ignoring any updates to it.